File: social-recipe-app-main/backend/src/main/java/com/bluepal/service/interfaces/ShoppingListService.java
package com.bluepal.service.interfaces;

import com.bluepal.entity.Recipe;
import com.bluepal.entity.ShoppingListItem;
import com.bluepal.entity.User;

import java.util.List;

public interface ShoppingListService {
    ShoppingListItem addItem(User user, String name, String quantity, String unit, Recipe recipe);
    List<ShoppingListItem> getItems(User user);
    ShoppingListItem togglePurchased(Long id, User user);
    void deleteCheckedItems(User user);
    void addIngredientsFromRecipe(Long recipeId, User user);
    void addIngredientsFromMealPlan(User user, java.time.LocalDate startDate, java.time.LocalDate endDate);
}

File: social-recipe-app-main/frontend/src/services/shoppingList.service.ts
import api from './api';

export interface ShoppingListItem {
  id: number;
  name: string;
  quantity: string;
  unit: string;
  category: string;
  recipeId?: number;
  recipeTitle?: string;
  purchased: boolean;
}

export interface ShoppingListItemRequest {
  name: string;
  quantity?: string;
  unit?: string;
}

export const shoppingListService = {
  getItems: () =>
    api.get<ShoppingListItem[]>(`/shopping-list`).then(r => r.data),
  
  addItem: (request: ShoppingListItemRequest) =>
    api.post<ShoppingListItem>(`/shopping-list`, request).then(r => r.data),
  
  togglePurchased: (id: number) =>
    api.patch<ShoppingListItem>(`/shopping-list/${id}/toggle`).then(r => r.data),
  
  deleteChecked: () =>
    api.delete(`/shopping-list/checked`).then(r => r.data),
  
  addFromRecipe: (recipeId: number) =>
    api.post(`/shopping-list/from-recipe/${recipeId}`).then(r => r.data),

  addFromMealPlan: (startDate: string, endDate: string) =>
    api.post(`/shopping-list/from-meal-plan`, null, { params: { startDate, endDate } }).then(r => r.data),
};
